### PR TITLE
[terra-application-v2] Upgrade to terra functional testing

### LINKED ISSUE #125: [terra-application-v2] Story Upgrade to use terra-functional-testing
# Story

## Description
As a maintainer of terra-application, I would like to upgrade the package to use [terra-functional-testing](https://github.com/cerner/terra-toolkit/tree/main/packages/terra-functional-testing).

## Motivation
So that the project can remain tech current on wdio which will also enable us to remain tech current with node in the long run

## Definition of Done
We'll know we're done when:
- A PR is open to master referencing the snapshot of [terra-functional-testing](https://github.com/cerner/terra-toolkit/tree/main/packages/terra-functional-testing)
- All tests are passing

### PR DESCRIPTION
**Pull request by mjhenkes**
### Summary

This pr upgrades us to terra-functional-testing and as a bonus caches docker images in travis ci.

Closes #125

### Deployment Link

https://terra-applic-.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -79,7 +79,7 @@
     "eventemitter3": "^4.0.4",
     "focus-trap-react": "^6.0.0",
     "intl": "^1.2.5",
-    "intl-locales-supported": "^1.8.4",
+    "intl-locales-supported": ">=1.8.4 <1.8.12 || ^1.8.13",
```
Comment (mjhenkes) [inline package.json:82]: this crazy version is to avoid a bad publish, version 1.8.12

File: package.json [hunk 2]
```diff
@@ -117,8 +117,7 @@
   },
   "devDependencies": {
     "@babel/cli": "^7.8.4",
-    "@babel/core": "7.10.5",
-    "@babel/parser": "7.10.5",
+    "@babel/core": "^7.10.5",
```
Comment (mjhenkes) [inline package.json:120]: unlocked this, the was locked to avoid a bug with the old react doc gen version in dev site v6

File: package.json [hunk 3]
```diff
@@ -149,16 +149,13 @@
     "html-webpack-plugin": "^4.5.0",
     "identity-obj-proxy": "^3.0.0",
     "jest": "^24.8.0",
-    "pacote": "^11.1.11",
```
Comment (mjhenkes) [inline package.json:152]: unused

File: package.json [hunk 4]
```diff
@@ -149,16 +149,13 @@
...
     "postcss": "^8.2.6",
-    "raf": "^3.4.1",
```
Comment (mjhenkes) [inline package.json:154]: unused

**Review by tbiethman — APPROVED** [review]

**Review by StephenEsser — APPROVED** [review]
This looks like it was a fairly smooth upgrade. Starting new test files in v2 probably helped.

File: .travis.yml [hunk 5]
```diff
@@ -34,27 +42,22 @@ jobs:
...
-        - docker-compose up -d standalone-chrome-travis
         # using the travis env section to set env variables seems to break cache
         - SITE=$TRAVIS_BUILD_DIR/travis-build/default npm run wdio-default
     - name: default-locale
       before_install: skip
       install: skip
       script:
-        - docker-compose up -d standalone-chrome-travis
-        - SITE=$TRAVIS_BUILD_DIR/travis-build/default-locale LOCALE=de npm run wdio-default
+        - SITE=$TRAVIS_BUILD_DIR/travis-build/default-locale run wdio-locale
```
Comment (mjhenkes) [inline .travis.yml]:
```suggestion
        - SITE=$TRAVIS_BUILD_DIR/travis-build/default-locale npm run wdio-locale
```

**Comment by mjhenkes** [comment]
> This looks like it was a fairly smooth upgrade. Starting new test files in v2 probably helped.

it’s easy to upgrade when you removed most of the tests in a previous pr 🤪

**Pull request merged**